lld/ELF/LinkerScript.cpp

```diff
@@ -406,12 +406,25 @@ static inline StringRef getFilename(const InputFile *file) {
   return file ? file->getNameForScript() : StringRef();
 }
 
+static inline StringRef getArchiveName(const InputFile *file) {
+  return file ? static_cast<StringRef>(file->archiveName) : StringRef();
```

Comment:
`StringRef(file->archiveName)`

https://en.cppreference.com/w/cpp/language/static_cast you were using the unnecessarily complex "If target-type is a reference type, the effect is the same as performing the declaration and initialization target-type temp(expression ﻿); "

Reply:
Actually, `file` is guaranteed to be non-nullptr. I am refactoring the caller to take a reference instead of a pointer to make this clear. Just delete this function

Reply:
Thanks for reviewing, I also went ahead and removed `getFilename` since it serves the same purpose.